R/zchunk_L244.building_USA.R

```diff
@@ -0,0 +1,917 @@
+      gather(share_type, share, share_tech1, share_tech2)%>%
+      gather(tech_type, technology, technology1, technology2) %>%
+      # Filter for same technology and share number, then remove tech_type and share_type columns
+      filter(substr(tech_type, nchar(tech_type), nchar(tech_type)) == substr(share_type, nchar(share_type), nchar(share_type))) %>%
+      select(-tech_type, -share_type)
+
+    # For calibration table, start with global tech efficiency table, repeat by states, and match in tech shares.
+    L244.StubTechCalInput_bld_gcamusa <- L244.GlobalTechEff_bld %>%
+      filter(year %in% BASE_YEARS) %>%
+      repeat_add_columns(tibble(region = gcamusa.STATES)) %>%
+      rename(supplysector = sector.name, subsector = subsector.name, stub.technology = technology) %>%
+      # Using left_join because we don't have shares for all technologies, NAs will be set to 1
+      left_join(L244.globaltech_shares, by = c("supplysector", "subsector", "stub.technology" = "technology")) %>%
+      replace_na(list(share = 1)) %>%
+      # Add energy by state/service/fuel
+      left_join_error_no_match(L244.in_EJ_R_bld_serv_F_Yh, by = c("region", "supplysector", "subsector", "minicam.energy.input", "year")) %>%
+      # calibrated.value = energy * share
+      mutate(calibrated.value = round(share * calibrated.value, energy.DIGITS_CALOUTPUT),
+             share.weight.year = year,
+             calOutputValue = calibrated.value) %>%
+      # Set subsector and technology shareweights
+      set_subsector_shrwt() %>%
+      mutate(tech.share.weight =  if_else(calibrated.value > 0, 1, 0)) %>%
+      select(LEVEL2_DATA_NAMES[["StubTechCalInput"]])
+
+    # L244.GlobalTechShrwt_bld_gcamusa: Default shareweights for global building technologies
+    L244.GlobalTechShrwt_bld_gcamusa <- A44.globaltech_shrwt %>%
+      gather(year, share.weight, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year)) %>%
+      complete(nesting(supplysector, subsector, technology), year = c(year, MODEL_YEARS)) %>%
+      group_by(supplysector, subsector, technology) %>%
+      mutate(share.weight = approx_fun(year, share.weight)) %>%
+      ungroup() %>%
+      filter(year %in% MODEL_YEARS) %>%
+      rename(sector.name = supplysector,
+             subsector.name = subsector) %>%
+      select(LEVEL2_DATA_NAMES[["GlobalTechYr"]], share.weight)
+
+    # L244.GlobalTechInterpTo_bld: Technology shareweight interpolation (selected techs only)
+    L244.GlobalTechInterpTo_bld <- A44.globaltech_interp %>%
+      mutate(sector.name = supplysector,
+             subsector.name = subsector) %>%
+      select(LEVEL2_DATA_NAMES[["GlobalTechInterpTo"]])
+
+    # L244.GlobalTechCost_bld: Non-fuel costs of global building technologies
+    L244.GlobalTechCost_bld_gcamusa <- A44.globaltech_cost %>%
+      gather(year, input.cost, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year)) %>%
```

Comment:
Replace these new lines with call to new `gather_year` function (merged in #821 )